### DISCUSSION TITLE
dzsave and icc_transform -  ICC profile correction vs skip-blanks vs file size

### DISCUSSION THREAD

**Discussion by delta2golf** [libvips/libvips#4297] [General]
Inconsistent behavior with SVS files: ICC profile correction vs skip-blanks vs file size

I'm encountering inconsistent behavior when processing SVS files with different combinations of ICC profile correction and skip-blanks options. 

Desired outcomes:
1. Skip blank areas
2. Correct ICC profile
3. Maintain reasonable output file size

Current Behavior

Command 1 - Skip blanks, no ICC correction:
vips dzsave wsi.svs[level=1] output.zip --skip-blanks 150 --background "255"
**ISSUE:** ICC not corrected
Output size: 7MB (not GB)

Command 2 - With ICC profile correction:
vips dzsave wsi.svs[level=1] output.zip --skip-blanks 150 --background "255" --profile srgb
**ISSUE**: Output size: 9 GB !!!

Command 3 - Using icc_transform:
vips icc_transform wsi.svs[level=1] output.dz srgb
**ISSUE:** Blanks: Not  skipped
Output size: 14MB (not GB)

**Comment by jcupitt** [comment]
Hello @delta2golf,

You need openslide4 for ICC profile support, it sounds like you might still be on 3.4.1

With openslide4 I see:

```
$ vipsheader -a CMU-1.svs | grep -i icc
aperio.ICC Profile: ScanScope v1
openslide.icc-size: 141992
icc-profile-data: 141992 bytes of binary data
```

I see:

```
$ vips dzsave CMU-1.svs[level=2] output.zip --skip-blanks 150 --background 255 --keep icc
$ unzip -qq output.zip
$ vipsheader -a output_files/5/0_0.jpeg 
0_0.jpeg: 23x17 uchar, 3 bands, srgb, jpegload
width: 23
height: 17
bands: 3
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 2.83465
yres: 2.83465
filename: 0_0.jpeg
vips-loader: jpegload
jpeg-multiscan: 0
jpeg-chroma-subsample: 4:2:0
icc-profile-data: 141992 bytes of binary data
```

Obviously 150kb of data on every tile is NOT a good idea. I would transform to sRGB with:

```
vips icc_transform CMU-1.svs[level=2] output.dz[skip-blanks=150,background=255] srgb
```

**Reply by jcupitt** [reply]
Oh, maybe you do have openslide4? Anyway, dzsave will strip profiles by default, it's much better to force to sRGB.
